lib/Targets/AIETargetXAIEV2.cpp

```diff
@@ -725,24 +725,44 @@ mlir::LogicalResult AIETranslateToXAIEV2(ModuleOp module, raw_ostream &output) {
     }
 
     for (auto connectOp : b.getOps<ConnectOp>()) {
-      if (connectOp.getSourceBundle() == WireBundle::North)
-        // demux!
-        output
-            << "__mlir_aie_try(XAie_EnableAieToShimDmaStrmPort("
-            << deviceInstRef << ", " << tileLocStr("x", "y")
-            << ", "
-            //               <<
-            //               stringifyWireBundle(connectOp.sourceBundle()).upper()
-            << connectOp.sourceIndex() << "));\n";
-      else if (connectOp.getDestBundle() == WireBundle::North)
-        // mux
-        output
-            << "__mlir_aie_try(XAie_EnableShimDmaToAieStrmPort("
-            << deviceInstRef << ", " << tileLocStr("x", "y")
-            << ", "
-            //               <<
-            //               stringifyWireBundle(connectOp.sourceBundle()).upper()
-            << connectOp.destIndex() << "));\n";
+
+      if (connectOp.getSourceBundle() == WireBundle::DMA ||
+          connectOp.getDestBundle() == WireBundle::DMA) {
+        if (connectOp.getSourceBundle() == WireBundle::North)
+          // demux!
+          output
+              << "__mlir_aie_try(XAie_EnableAieToShimDmaStrmPort("
+              << deviceInstRef << ", " << tileLocStr("x", "y")
+              << ", "
+              //               <<
+              //               stringifyWireBundle(connectOp.sourceBundle()).upper()
+              << connectOp.sourceIndex() << "));\n";
+        else if (connectOp.getDestBundle() == WireBundle::North)
+          // mux
+          output
+              << "__mlir_aie_try(XAie_EnableShimDmaToAieStrmPort("
+              << deviceInstRef << ", " << tileLocStr("x", "y")
+              << ", "
+              //               <<
+              //               stringifyWireBundle(connectOp.sourceBundle()).upper()
+              << connectOp.destIndex() << "));\n";
+      }
+
+      else if (connectOp.getSourceBundle() == WireBundle::PLIO ||
+               connectOp.getDestBundle() == WireBundle::PLIO) {
+        // Note: Right now this just works with PLIO channel 0 and 1 as those
+        // don't require to program the shim mux
+        if (connectOp.destIndex() != 0 && connectOp.destIndex() != 1) {
+          connectOp.emitWarning(
+              "Currently only PLIO channel 0 and 1 are supported.");
+        }
+
+        if (connectOp.getDestBundle() == WireBundle::North)
+          // mux
+          output << "__mlir_aie_try(XAie_PlToAieIntfEnable(" << deviceInstRef
+                 << ", " << tileLocStr("x", "y") << ", "
+                 << connectOp.destIndex() << ", PLIF_WIDTH_64));\n";
+      }
```

Comment:
What about the case of being connected TO a PLIO?

Reply:
Currently we are only supporting the 0th and 1st channel as mentioned in https://github.com/Xilinx/mlir-aie/blob/e05bff488e5c3fc940334b0da7a603d0106638f6/lib/Targets/AIETargetXAIEV2.cpp#L757. For AIE -> PL using those channels it is a straightshot to the PL and we actually don't need to configure anything. From PL -> AIE we need a little configuration which is what this line is doing. This is all tested in hardware.

Reply:
OK, would like to see the proper configuration happen here, since there's no fundamental reason why the other cases can't be implemented as well.  At the very least, this should be an error rather than a warning.

Reply:
Good point, it was originally an error but that made one of the plio unit tests fail. The reason that I didn't put them in there is that it is difficult to test as I am not sure how at platform build time to force a PLIO connection to use a specific channel. I will look into it and if that doesn't work, I will at least make it an error and change the unit test that was failing.

Reply:
[0f50197](https://github.com/Xilinx/mlir-aie/pull/1623/commits/0f501974e7a6e94b4a43e1cfd77405ad81a06fb3) Removed the channel limitation in this commit.